### PR TITLE
Merge index checker annotations into plume-lib proper

### PR DESCRIPTION
**Pull request by kelloggm**
@mernst, I don't expect this PR to be merged right away, but I thought it would be an easier way to communicate about the state of the plume-lib case study. I've gone through the whole thing and I think the annotations, at least, are (nearly) ready to be merged. I've dealt with all of the "Index TODO" comments and replaced them either with helpful messages about why warnings were being suppressed or removed them entirely. I'd appreciate it if you'd take another look at the annotations and let me know if there's anything else that concerns you. If not, then we can start the process to actually get these merged :)

### CODE REVIEW

File: java/lib/bcel.astub [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import org.checkerframework.checker.index.qual.SameLen;
```
Comment (mernst) [inline java/lib/bcel.astub]: This stub file should appear in the Checker Framework, not in plume-lib

File: java/src/plume/LimitedSizeSet.java [hunk 2]
```diff
@@ -45,17 +49,23 @@
...
     if (assertsEnabled && !(max_values > 0)) {
       throw new IllegalArgumentException("max_values should be positive, is " + max_values);
     }
     // this.max_values = max_values;
-    @SuppressWarnings("unchecked")
-    /*@Nullable*/ T[] new_values_array = (/*@Nullable*/ T[]) new /*@Nullable*/ Object[max_values];
+    @SuppressWarnings({
+      "unchecked",
+      "index", // https://github.com/kelloggm/checker-framework/issues/174
+      "value"
```
Comment (mernst) [inline java/src/plume/LimitedSizeSet.java]: This is a new `@SuppressWarnings` (for "value"), but there is no comment about it.

File: java/lib/bcel.astub (hunk 1 above)
Comment (kelloggm) [reply java/lib/bcel.astub]: That makes sense. I'll do that.

Comment (kelloggm) [reply java/lib/bcel.astub]: See https://github.com/typetools/commons-bcel/pull/4

**Comment by mernst** [comment]
This is blocked on typetools/checker-framework#1613

**Comment by mernst** [comment]
Though the build failure seems to be due to a timeout, so typetools/checker-framework#1631 should help too.

**Comment by kelloggm** [comment]
There shouldn't be any difference. I'll do it.

**Comment by mernst** [comment]
Also, looking at the warning suppressions, these two issues have been closed so those warning suppressions can be removed:
       https://github.com/kelloggm/checker-framework/issues/175
       https://github.com/kelloggm/checker-framework/issues/172
I'll do that too, in the upstream plume-lib-check-index-new

File: java/src/plume/BCELUtil.java [hunk 3]
```diff
@@ -494,8 +496,14 @@ public static void add_line_numbers(MethodGen mg, InstructionList il) {
   public static void setup_init_locals(MethodGen mg) {
 
     // Get the parameter types and names.
-    Type[] arg_types = mg.getArgumentTypes();
-    String[] arg_names = mg.getArgumentNames();
+    @SuppressWarnings(
+        "nullness") // The initialization checker warns here when the SameLen annotations are added.
```
Comment (mernst) [inline java/src/plume/BCELUtil.java]: Can you make the comment explain why the code is correct?  (Same comment a few other places.)

**Comment by mernst** [comment]
I'm seeing some warnings when I run the Index Checker:

```
src/plume/ArraysMDE.java:546: warning: [argument.type.incompatible] incompatible types in argument.
      if (elt.equals(a.get(i))) {
                           ^
  found   : @LowerBoundUnknown int
  required: @NonNegative int
src/plume/ArraysMDE.java:638: warning: [argument.type.incompatible] incompatible types in argument.
      if (elt == a.get(i)) {
                       ^
  found   : @LowerBoundUnknown int
  required: @NonNegative int
error: warnings found and -Werror specified
src/plume/StringBuilderDelimited.java:93: warning: [return.type.incompatible] incompatible types in return.
    return delegate.length();
                          ^
  found   : @LowerBoundUnknown int
  required: @NonNegative int
1 error
3 warnings
```

Can you resolve those, or at least add a suppression pointing at the relevant issue?

File: .travis-build.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (mernst) [inline .travis-build.sh]: This file still needs to be removed, I believe.

**Pull request merged**